File: docs/reference/OpenViking/openviking_cli/cli/commands/resources.py
# Copyright (c) 2026 Beijing Volcano Engine Technology Co., Ltd.
# SPDX-License-Identifier: Apache-2.0
"""Resource management commands."""

from typing import Optional

import typer

from openviking_cli.cli.errors import run


def register(app: typer.Typer) -> None:
    """Register resource commands."""

    @app.command("add-resource")
    def add_resource_command(
        ctx: typer.Context,
        path: str = typer.Argument(..., help="Local path or URL to import"),
        to: Optional[str] = typer.Option(None, "--to", help="Target URI"),
        reason: str = typer.Option("", help="Reason for import"),
        instruction: str = typer.Option("", help="Additional instruction"),
        wait: bool = typer.Option(False, "--wait", help="Wait until processing is complete"),
        timeout: Optional[float] = typer.Option(600.0, help="Wait timeout in seconds"),
    ) -> None:
        """Add resources into OpenViking."""
        run(
            ctx,
            lambda client: client.add_resource(
                path=path,
                target=to,
                reason=reason,
                instruction=instruction,
                wait=wait,
                timeout=timeout,
            ),
        )

    @app.command("add-skill")
    def add_skill_command(
        ctx: typer.Context,
        data: str = typer.Argument(..., help="Skill directory, SKILL.md, or raw content"),
        wait: bool = typer.Option(False, "--wait", help="Wait until processing is complete"),
        timeout: Optional[float] = typer.Option(600.0, help="Wait timeout in seconds"),
    ) -> None:
        """Add a skill into OpenViking."""
        run(
            ctx,
            lambda client: client.add_skill(
                data=data,
                wait=wait,
                timeout=timeout,
            ),
        )

File: docs/reference/OpenViking/crates/ov_cli/src/commands/resources.rs
use crate::client::HttpClient;
use crate::error::Result;
use crate::output::{output_success, OutputFormat};

pub async fn add_resource(
    client: &HttpClient,
    path: &str,
    to: Option<String>,
    reason: String,
    instruction: String,
    wait: bool,
    timeout: Option<f64>,
    format: OutputFormat,
    compact: bool,
) -> Result<()> {
    let result = client
        .add_resource(path, to, &reason, &instruction, wait, timeout)
        .await?;
    output_success(&result, format, compact);
    Ok(())
}

pub async fn add_skill(
    client: &HttpClient,
    data: &str,
    wait: bool,
    timeout: Option<f64>,
    format: OutputFormat,
    compact: bool,
) -> Result<()> {
    let result = client.add_skill(data, wait, timeout).await?;
    output_success(&result, format, compact);
    Ok(())
}
